### PR TITLE
Issue 329: version 4 nodes cannot connect to version 3

### PR DESCRIPTION
**Pull request by mgoulish**
* When encountering secret that has no 'generated-by'
  annotation, issue warning but do not fail.

### CODE REVIEW

File: client/connector_create.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func secretFileAuthor(ctx context.Context, secretFile string) (author string, er
 	}
 	generatedBy, ok := content["skupper.io/generated-by"]
 	if !ok {
-		return "", fmt.Errorf("Can't find secret origin.")
```
Comment (ajssmith) [inline client/connector_create.go:53]: Suggest leaving as is and caller checks for error strings.

File: client/connector_create.go [hunk 2]
```diff
@@ -90,10 +90,15 @@ func (cli *VanClient) ConnectorCreateFromFile(ctx context.Context, secretFile st
 	// secrets that we have used to make connections.
 	newConnectionAuthor, err := secretFileAuthor(ctx, secretFile)
 	if err != nil {
-
 		return nil, err
 	}
 
+	alreadyWarned := false
+	if newConnectionAuthor == "unknown" {
```
Comment (ajssmith) [inline client/connector_create.go:97]: Suggest no using a magic string and check for error response instead.

File: client/connector_create.go [hunk 3]
```diff
@@ -102,7 +107,10 @@ func (cli *VanClient) ConnectorCreateFromFile(ctx context.Context, secretFile st
 	for _, oldSecret := range secrets.Items {
 		oldConnectionAuthor, ok := oldSecret.Annotations["skupper.io/generated-by"]
 		if !ok {
-			return nil, fmt.Errorf("A secret has no author.")
+			if !alreadyWarned {
```
Comment (ajssmith) [inline client/connector_create.go:110]: Lean towards not generating a log:

1. The client pattern is to return error indication and leave it to the caller to generate an indication or prompt to the user
2. With the 0.3 release, a user can make multiple connections "silently"

**Review by ajssmith — COMMENTED** [review]
iiuc, the goal of the change is to make it easier for 0.4 sites to connect to 0.3 sites (e.g. not require them to annotate the connection  tokens generated from a 0.3 site). As it should be a short term transition, it would be ok to just let the connections complete "silently" and turn on enforcement in the 0.5 release. 

Thoughts?

**Comment by mgoulish** [comment]
withdrawn in favor of PR 331

**Pull request closed without merging**